lib/screens/v3_api/parser.ex

```diff
@@ -0,0 +1,109 @@
+defmodule Screens.V3Api.Parser do
+  @moduledoc "Tools for parsing V3 API responses."
+
+  @parsers %{
+    "alert" => Screens.Alerts.Parser,
+    "facility" => Screens.Facilities.Parser,
+    "prediction" => Screens.Predictions.Parser,
+    "route" => Screens.Routes.Parser,
+    "route_pattern" => Screens.RoutePatterns.Parser,
+    "schedule" => Screens.Schedules.Parser,
+    "stop" => Screens.Stops.Parser,
+    "trip" => Screens.Trips.Parser,
+    "vehicle" => Screens.Vehicles.Parser
+  }
+
+  @typep included :: %{{id :: String.t(), type :: String.t()} => object()}
+  @typep object :: %{String.t() => term()}
+
+  @doc "Parse a complete JSON-decoded V3 API response."
+  @spec parse(response :: object()) :: struct() | [struct()]
+  def parse(%{"data" => data} = response) when is_map(data) do
+    [resource] = parse(%{response | "data" => [data]})
+    resource
+  end
+
+  def parse(%{"data" => data} = response) when is_list(data) do
+    included =
+      response
+      |> Map.get("included", [])
+      |> then(&Enum.concat(data, &1))
```

Comment:
This added line (versus the previously duplicated copies of this logic in the per-resource parsers) fixes [a crash we've seen](https://mbtace.sentry.io/issues/6492234261) on the test on-bus screen. This occurred when trying to resolve a relationship that referred to a resource in the response's `data` (not `included`).

Reply:
great catch and fix idea by moving everything into a shared parser so this logic can be addressed everywhere!